src/Microsoft.TestPlatform.Common/ExtensionFramework/VSExtensionManager.cs

```diff
@@ -0,0 +1,203 @@
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.Common.ExtensionFramework
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.IO;
+    using System.Reflection;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+    using Microsoft.VisualStudio.TestPlatform.Utilities.Helpers.Interfaces;
+    using Microsoft.VisualStudio.TestPlatform.Utilities.Helpers;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.Client;
+    using Microsoft.VisualStudio.TestPlatform.PlatformAbstractions;
+    using Microsoft.VisualStudio.TestPlatform.Common.Utilities;
+    using Microsoft.VisualStudio.TestPlatform.Common.Interfaces;
+    using Microsoft.VisualStudio.TestPlatform.Common.Resources;
+
+    public class VSExtensionManager : IVSExtensionManager
+    {
+        private const string PrivateAssembliesDirName = "PrivateAssemblies";
+        private const string PublicAssembliesDirName = "PublicAssemblies";
+        private const string ExtensionManagerService = "Microsoft.VisualStudio.ExtensionManager.ExtensionManagerService";
+        private const string ExtensionManagerAssemblyName = @"Microsoft.VisualStudio.ExtensionManager";
+        private const string ExtensionManagerImplAssemblyName = @"Microsoft.VisualStudio.ExtensionManager.Implementation";
+        private const string DevenvExe = "devenv.exe";
+
+        private const string SettingsManagerTypeName = "Microsoft.VisualStudio.Settings.ExternalSettingsManager";
+        private const string SettingsManagerAssemblyName = @"Microsoft.VisualStudio.Settings.15.0";
+        
+        private IFileHelper fileHelper;
+        private string vsInstallPath;
+        private string pathToDevenv;
+
+        private Assembly extensionManagerAssembly;
+        private Assembly extensionManagerImplAssembly;
+        private Type extensionManagerServiceType;
+
+        private Assembly settingsManagerAssembly;
+        private Type settingsManagerType;
+
+        public VSExtensionManager() : this(new FileHelper())
+        {
+        }
+
+        internal VSExtensionManager(IFileHelper fileHelper)
+        {
+            this.fileHelper = fileHelper;
+        }
+
+        /// <summary>
+        /// Get the available unit test extensions installed.
+        /// If no extensions are installed then it returns an empty list.
+        /// </summary>
+        /// <returns></returns>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1024:UsePropertiesWhereAppropriate"), System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        public IEnumerable<string> GetUnitTestExtensions()
+        {
+            try
+            {
+                return GetTestExtensionsInternal(Constants.UnitTestExtensionType);
```

Comment:
Is there a benefit of an extra method call here?

Reply:
Just looks cleaner, and YAGNI maybe, but thought if we need other extension types, we don't need to refactor.